storage-provider/common/src/deadline.rs

```diff
@@ -0,0 +1,203 @@
+use std::{collections::BTreeSet, sync::Arc};
+
+use polka_storage_proofs::{
+    porep::sealer::{BlstrsProof, SubstrateProof},
+    post::{self, PoStParameters, ReplicaInfo},
+};
+use primitives::{
+    proofs::{derive_prover_id, RegisteredPoStProof},
+    randomness::{draw_randomness, DomainSeparationTag},
+    sector::SectorNumber,
+};
+use storagext::{
+    runtime::runtime_types::primitives::pallets::DeadlineInfo,
+    types::storage_provider::{PartitionState, PoStProof, SubmitWindowedPoStParams},
+    RandomnessClientExt, StorageProviderClientExt, SystemClientExt,
+};
+use subxt::{ext::codec::Encode, tx::Signer};
+use tokio::task::{JoinError, JoinHandle};
+
+use crate::sector::ProvenSector;
+
+#[derive(Debug, thiserror::Error)]
+pub enum DeadlineError {
+    #[error("precommit scheduled too early, randomness not available")]
+    RandomnessNotAvailable,
+    #[error("current deadline or storage provider not found")]
```

Comment:
```suggestion
    #[error("current deadline for storage provider not found")]
```
NIT: Looking at the usage of this error it is not really the current deadline that is not found. This error is emitted when looking for a deadline at a specific index.

Reply:
This is also failing if the account id you provided is not a storage provider and you're trying to get the deadline info for it.